ci/Jenkinsfile

```
@@ -1,39 +1,133 @@
-pipeline {
-    agent { label 'permaplant' }
 
-    stages {
-        stage('Build PR') {
-            environment {
-                DATABASE_URL = 'postgres://cidb:cidb@127.0.0.1/cidb'
+stage('Build Schema'){
+    node('docker') {
+        checkout scm
+        def rust_image = docker.build("permaplant-rust", "./ci/container-images/permaplant-rust")
+
+        docker.image('postgres:alpine').withRun('-e "POSTGRES_USER=ci" -e "POSTGRES_PASSWORD=ci"') { c ->
+            rust_image.inside("--link ${c.id}:db -e 'DATABASE_URL=postgres://ci:ci@db/ci'") {
+                checkout scm
+                sh './ci/build-scripts/build-schema.sh'
+
+                stash includes: 'backend/src/schema.rs', name: 'schema.rs'
+                stash includes: 'frontend/src/bindings/definitions.ts', name: 'definitions.ts'
             }
+        }
+    }
+}
 
-            steps {
-                echo "Workspace is '${WORKSPACE}'"
+stage('Checks') {
+    def parallel_checks = [:]
 
-                echo 'Ensure pr service is stopped'
-                sh 'sudo /usr/sbin/service permaplant-pr stop'
+    parallel_checks["check-backend"] = {
+        node('docker') {
+            checkout scm
+            def rust_image = docker.build("permaplant-rust", "./ci/container-images/permaplant-rust")
 
-                echo 'Resetting the CI DB ...'
-                sh 'sudo /usr/local/bin/permaplant-reset-ci.sh'
+            docker.image('postgres:alpine').withRun('-e "POSTGRES_USER=ci" -e "POSTGRES_PASSWORD=ci"') { c ->
+                rust_image.inside("--link ${c.id}:db -e 'DATABASE_URL=postgres://ci:ci@db/ci'") {
+                    checkout scm
+                    sh './ci/build-scripts/build-schema.sh'
 
-                echo 'Building...'
-                sh './ci/build-scripts/build-backend.sh'
-                sh './ci/build-scripts/build-frontend.sh'
+                    dir("backend") {
+                        withEnv(['RUSTFLAGS=-D warnings', 'RUSTDOCFLAGS=-D warnings']) {
+                            sh "cargo fmt --check"
+                            sh "cargo check"
+                            sh "cargo clippy"
+                            sh "cargo doc"
+                            withEnv(['BIND_ADDRESS_HOST=127.0.0.1', 'BIND_ADDRESS_PORT=8080']) {
+                                sh "cargo test"
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
 
-                echo 'Deploying'
-                sh "sudo /usr/local/bin/permaplant-deploy.sh pr ${WORKSPACE}"
+    parallel_checks["check-frontend"] = {
+        node('docker') {
+            checkout scm
+            def node_image = docker.build("permaplant-node", "./ci/container-images/permaplant-node")
+
+            node_image.inside {
+                checkout scm
+                unstash 'definitions.ts'
+
+                echo "HC SVNT DRACONES"
```

As soon as https://github.com/ElektraInitiative/PermaplanT/pull/171 is merged you can add these changes.
```suggestion
                sh "npm run format:check"
                sh "npm run lint"
                sh "npm run test"
```

---

as these has dependencies on other PRs, I will do this later.
We need the backend checks asap in master :)